### PR TITLE
PERF-4017 Add subset of QE workloads to Genny

### PR DESCRIPTION
**Pull request by spencerjackson**
Thanks for submitting a PR to the Genny repo. Please include the following fields (if relevant) prior to submitting your PR.

**Jira Ticket:** PERF-4017

**Whats Changed:**  
This patch commits the test generator scripts we used to perform acceptance testing for Queryable Encryption, as well as a subset of the tests that we'll want to run continuously after [PERF-3954](https://jira.mongodb.org/browse/PERF-3954).

**Patch testing results:**  
Raw patch: https://spruce.mongodb.com/version/64383af7e3c331d3b0efedff/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC
As is, this runs no new workloads because our QE workloads are currently unable to run without a newer version of libmongocrypt in the Genny toolchain. An upgrade to libmongocrypt, which would allow the added YAML workloads to run is described in [PERF-3954](https://jira.mongodb.org/browse/PERF-3954).

Here is a variant of the patch, which ran a modified Genny toolchain: https://spruce.mongodb.com/version/643813d632f417f30515a464/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

**Workload Submission form:**  
If applicable, only required if there is a new workload being added. Form can be found [here](https://docs.google.com/forms/d/1r0kOHvFUa7rJxVmX_wp9prxYU5K155yKyZ1y3d5yhZg/edit)

**Related PRs:**   
If applicable, link related PRs
N/A.

### CODE REVIEW

**Review by rtimmons — APPROVED** [review]
Love the contrib stuff! I poked through and didn't see anything super obvious. In general the `contrib` directory is pretty free-for-all/self-service as are the actual workloads, so as long as it works for your purposes please go wild. (Always happy to poke if you want additional eyes.)

File: src/workloads/encrypted/medical_workload-load-unencrypted.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+#       - shard-lite-fle
+#     branch_name:
+#       $neq:
+#       - v4.0
+#       - v4.2
+#       - v4.4
+#       - v5.0
+#       - v6.0
+#       - v6.1
+#       - v6.2
```
Comment (erwee) [inline src/workloads/encrypted/medical_workload-load-unencrypted.yml:102]: no v6.3?

Comment (spencerjackson) [reply src/workloads/encrypted/medical_workload-load-unencrypted.yml]: Hah. I'd just copied this from the other tests. OK. I've reworked this to use `$gte: 7.0`. How does that look?

**Review by erwee — APPROVED** [review]

File: src/workloads/encrypted/medical_workload-load-unencrypted.yml (hunk 1 above)
Comment (erwee) [reply src/workloads/encrypted/medical_workload-load-unencrypted.yml]: lgtm!

**Comment by spencerjackson** [comment]
evergreen merge

**Pull request merged**